expression/builtin_string.go

```diff
@@ -424,33 +424,35 @@ type lowerFunctionClass struct {
 }
 
 func (c *lowerFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinLowerSig{newBaseBuiltinFunc(args, ctx)}
+	tp := types.NewFieldType(mysql.TypeVarchar)
+	tp.Charset, tp.Collate = charset.CharsetUTF8, charset.CollationUTF8
+	argTp := args[0].GetType()
+	if types.IsBinaryStr(argTp) || argTp.Tp == mysql.TypeNull {
+		types.SetBinChsClnFlag(tp)
+	}
+
+	bf, err := newBaseBuiltinFuncWithTp(args, tp, ctx, tpString)
```

Comment:
Hi, since function `newBaseBuiltinFuncWithTp(...)` is refactored in this pull request [https://github.com/pingcap/tidb/pull/3617](https://github.com/pingcap/tidb/pull/3617), please update your code and refactor it again, sorry for the inconvenience